### PR TITLE
Image component in .NET SDK

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Moving from a fork https://github.com/pulumi/pulumi-docker/pull/115 to run the tests

### CODE REVIEW

File: sdk/dotnet/Docker.cs [hunk 1]
```diff
@@ -0,0 +1,699 @@
...
+                // In case it does, return an error with the exception printout.
+                return (1, ex.ToString());
+            }
+        }
+
+        private static async Task WaitForExitAsync(Process process, CancellationToken cancellationToken = default)
+        {
+            var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            void ProcessExited(object? sender, EventArgs e) => tcs.TrySetResult(true);
```
Comment (CyrusNajmabadi) [inline sdk/dotnet/Docker.cs]: can you place local functions att the end of the containing method.  also, place a `return;` statement above it.  it makes it very clear waht the separation is between teh normal code and helper functions.

File: sdk/dotnet/Docker.cs [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+            }
+            catch (Exception ex)
+            {
+                // This shouldn't normally happen, but we want to be sure Process doesn't throw.
+                // In case it does, return an error with the exception printout.
+                return (1, ex.ToString());
+            }
+        }
+
+        private static async Task WaitForExitAsync(Process process, CancellationToken cancellationToken = default)
```
Comment (CyrusNajmabadi) [inline sdk/dotnet/Docker.cs]: you neve pass in this cancellation token.  please remove.

Comment (mikhailshilkov) [reply sdk/dotnet/Docker.cs]: removed

File: sdk/dotnet/Docker.cs (hunk 1 above)
Comment (mikhailshilkov) [reply sdk/dotnet/Docker.cs]: done

**Pull request merged**